libdd-profiling-ffi/src/profile_status.rs

```diff
@@ -0,0 +1,487 @@
+// Copyright 2025-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::ProfileError;
+use allocator_api2::alloc::{AllocError, Allocator, Global, Layout};
+use std::borrow::Cow;
+use std::ffi::{c_char, CStr, CString};
+use std::fmt::Display;
+use std::mem::ManuallyDrop;
+use std::ptr::{null, NonNull};
+
+const FLAG_OK: usize = 0b00;
+const FLAG_STATIC: usize = 0b01;
+const FLAG_ALLOCATED: usize = 0b11;
+
+const MASK_IS_ERROR: usize = 0b01;
+const MASK_IS_ALLOCATED: usize = 0b10;
+const MASK_UNUSED: usize = !(MASK_IS_ERROR | MASK_IS_ALLOCATED);
```

Comment:
There is an implicit requirement that these bitfields stay in sync.  Is there a test for that?

Reply:
No, I will write one after the release 👍🏻